packages/valory/skills/abstract_round_abci/models.py

```diff
@@ -42,27 +56,178 @@
     get_name,
 )
 from packages.valory.skills.abstract_round_abci.utils import (
-    DEFAULT_TENDERMINT_P2P_URL,
     get_data_from_nested_dict,
     get_value_with_type,
 )
 
 
 NUMBER_OF_RETRIES: int = 5
-DEFAULT_BACKOFF_FACTOR: int = 2
+DEFAULT_BACKOFF_FACTOR: float = 2.0
 DEFAULT_TYPE_NAME: str = "str"
-_DEFAULT_REQUEST_TIMEOUT = 10.0
-_DEFAULT_REQUEST_RETRY_DELAY = 1.0
-_DEFAULT_TX_TIMEOUT = 10.0
-_DEFAULT_TX_MAX_ATTEMPTS = 10
-_DEFAULT_CLEANUP_HISTORY_DEPTH_CURRENT = None
 
 
 HeadersType = List[Tuple[str, str]]
 ParametersType = HeadersType
 
 
-class BaseParams(Model):  # pylint: disable=too-many-instance-attributes
+class FrozenMixin:  # pylint: disable=too-few-public-methods
+    """Mixin for classes to enforce read-only attributes."""
+
+    _frozen: bool = False
+
+    def __delattr__(self, *args: Any) -> None:
+        """Override __delattr__ to make object immutable."""
+        if self._frozen:
+            raise AttributeError("This object is frozen!")
+        super().__delattr__(*args)
+
+    def __setattr__(self, *args: Any) -> None:
+        """Override __setattr__ to make object immutable."""
+        if self._frozen:
+            raise AttributeError("This object is frozen!")
+        super().__setattr__(*args)
+
+
+class TypeCheckMixin:  # pylint: disable=too-few-public-methods
+    """Mixin for data classes to enforce attribute types on construction."""
+
+    def __post_init__(self) -> None:
```

Comment:
This is an incomplete implementation. We will have to extend it to cover more complex types.